src/sos_constraints.rs

```diff
@@ -0,0 +1,49 @@
+use std::os::raw::c_int;
+use std::convert::TryInto;
+use crate::raw::SOSConstraintType;
+use crate::Col;
+
+/// Represents a group of multiple special ordered set constraints
+#[derive(Clone, Debug)]
+pub(crate) struct SOSConstraints {
+    // It is important that these fields remain private in order to ensure
+    // memory safety of the API. They can only be mutated together through the public
+    // methods of this struct.
+    row_starts: Vec<c_int>,
+    col_indices: Vec<c_int>,
```

Comment:
At this level of abstraction, I'd prefer to manage Col objects, and only convert them to c_int at the last time.

Reply:
Only Col objects are exposed in the interface of SOSConstraints. The types of these private internal fields are an implementation detail, and changing them to Col would just force to iterate over them for nothing before being able to add them.

Reply:
OK